### ISSUE TITLE
Followers' and Featured Hashtags' counter numbers change based on web browser's locale language

### ISSUE THREAD

**Issue by KNTRO** [tootsuite/mastodon#16435]
### Expected behaviour

The quantity of followers and featured hashtags should be accurate always

### Actual behaviour

The quantity of followers and featured hashtags have variations depending on the web browser's locale

### Steps to reproduce the problem

1. Select a language for webpages in your web browser that also exist in Mastodon
2. Open a private web browsing session window of your web browser
3. Open your Mastodon profile and check your _Followers_ number and Featured Hashtag's number (if any)
4. Close that private web browsing session window
5. In your web browser's settings, select a language for webpages that does not exist in Mastodon
6. Open a new private web browsing session window
7. Open your Mastodon profile again and check your _Followers_ number and Featured Hashtag's number (if any)

**This is what I see in step _3_ (accurate numbers)**
![masOK](https://user-images.githubusercontent.com/2568821/123585754-18821800-d7ba-11eb-93bd-c35fd6332bc0.jpg)

**And this is what I see in step _7_ (wrong numbers)**
![masfailed](https://user-images.githubusercontent.com/2568821/123585758-19b34500-d7ba-11eb-9d4e-4bd6eda79358.jpg)

So, basically: **If the web browser's selected language for displaying webpages does not match any of the available Mastodon locales, then the numbers shown there are not accurate**.

### Specifications

Mastodon: Web, v3.4.1
OS: Windows 10 Home, Version 20H2, 64 bit
Web browser: Waterfox G3.2.3, 64 bit

**Comment by ClearlyClaire** [comment]
This is not an issue with a selected language not being supported by Mastodon, but an issue affecting some languages. I am trying to get to the root cause of this, but basically, we call a Rails method called `number_to_human` to display numbers in a short human-readable way. This method is dependent on language since different languages may have different usual separators, unit names and rules.

However, the way it rounds numbers for Spanish is pretty surprising. Here are Mastodon-supported locales where “197” do not get rendered as “197” but something else. I am not well-versed in any of them, but there is probably something off with them:

```
bg: 200
bn: 200
ca: 200
cs: 197,0
el: 200
es: 200
et: 200
eu: 200
fa: 200
gl: 200
hu: 200
lt: 200
lv: 197,0
ru: 197,0
sk: 197,0
sl: 200
sv: 200
uk: 197,0
ur: 197 Rs
vi: 200
zh-CN: 197.0
zh-HK: 197.0
zh-TW: 197.0
```

**Comment by ClearlyClaire** [comment]
This comes from `rails-i18n`'s defaults, which are [the following for `number.human.format` in the `en` locale](https://github.com/svenfuchs/rails-i18n/blob/70ac505eaf8e934c269c1f565c4e2e515b7b4d1d/rails/locale/en.yml#L180-L184):
```yaml
        delimiter: ''
        precision: 3
        significant: true
        strip_insignificant_zeros: true
```

And [the following in the `es` locale](https://github.com/svenfuchs/rails-i18n/blob/70ac505eaf8e934c269c1f565c4e2e515b7b4d1d/rails/locale/es.yml#L184-L188):
```yaml
        delimiter: ''
        precision: 1
        significant: true
        strip_insignificant_zeros: true
```

I am not sure what the rationale is for those discrepancies. We can overwrite them in the locale files or in the calls to `number_to_human`.

**Comment by KNTRO** [comment]
@ClearlyClaire Thanx a bunch for all this in-depth info, didn't know about this. Should be closed this issue or not?

**Comment by ClearlyClaire** [comment]
No, I don't think the issue should be closed. The precision setting for at least Spanish is confusing and probably not what we want. I think this needs to be fixed one way or another, and probably for all locales. I'm just not sure whether it's best to overwrite the settings in each locale file, or in each `number_to_human` call.

@Gargron opinion?

**Comment by nightpool** [comment]
I think it makes the most sense to override the precision at the call-site,
since the right amount of precision is going to change based on the
specific context
